src/components/reference-editor/index.js

```diff
@@ -0,0 +1,53 @@
+import React, { PropTypes } from 'react';
+import reactStamp from 'react-stamp';
+
+import EditorFactory from 'components/editor';
+
+import createMentionPlugin, { defaultSuggestionsFilter } from 'draft-js-mention-plugin';
+import createUndoPlugin from 'draft-js-undo-plugin';
+import createLinkifyPlugin from 'draft-js-linkify-plugin';
+
+import 'draft-js-mention-plugin/lib/plugin.css';
+import 'draft-js-undo-plugin/lib/plugin.css';
+import 'draft-js-linkify-plugin/lib/plugin.css';
+
+import MentionComponent from './mention-text';
+import SuggestionsFactory from './suggestions';
+
+const Editor = EditorFactory(React);
```

Comment:
The purpose of passing `React` as an argument to the factory is to ensure the same instance is used everywhere. When done properly, there isn't the need to `import React` at all. Let's move the call of this factory to inside the factory beneath so as to take advantage of that same instance.